aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -220,6 +220,38 @@ In the following example, the call to `AddSingleton` registers `MyDependency` as
 services.AddSingleton<IMyDependency, MyDependency>();
 // The following line has no effect:
 services.TryAddSingleton<IMyDependency, DifferentDependency>();
+
+public class MyService
+{
+    public MyService(IMyDependency myDependency, 
+        IEnumberable<IMyDependency> myDependencies)
+    {
+        Trace.Assert(myDependency is MyDependency);
+        Trace.Assert(myDependencies.Single() is MyDependency);
+    }
+}
+```
+
+`DifferentDependency` replaces `MyDependency` if the call to `TryAddSingleton` is changed to a second call to `AddSingleton`, unless multiple `IMyDependency` instances are resolved via `IEnumerable<IMyDependency>`. In that case, the second call to `AddSingleton` adds `DifferentDependency` after `MyDependency` in the `IEnumerable<IMyDependency>`.
```

Comment:
Is this bit needed if we immediately show the example below? I'm asking because that sentence is non-trivial to read.

Reply:
```suggestion
```